Generator/Sources/NeedleFramework/Models/Pluginized/PluginizedProperty.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// When a property is found is the auxillary source, this lets us know the source
+enum AuxillarySourceType {
+    /// Situation where a core component is getting a property from the plugin extension protocol
+    case pluginExtension
+    /// Situation where a non-core component gets the property from a parent non-core component
```

Comment:
Technically it's not always the `parent` right? It should be "ancestor`.

Reply:
done